lib/AST/Type.cpp

```diff
@@ -2285,7 +2285,7 @@ static bool matches(CanType t1, CanType t2, TypeMatchOptions matchMode,
 
   // Class-to-class.
   if (matchMode.contains(TypeMatchFlags::AllowOverride))
-    if (t2->isExactSuperclassOf(t1))
+    if (t2->eraseDynamicSelfType()->isExactSuperclassOf(t1))
```

Comment:
This wasn't necessary for method overrides, so it shouldn't be necessary to add this to make variable and subscript overrides work.

Reply:
This might be the same underlying problem as subscript overrides not working in generic classes

Reply:
Ah, figured this one out. You need to teach `TypeBase::adjustSuperclassMemberDeclType()` about SubscriptDecl and VarDecl returning Self too.

I feel like the bit we want to factor out here that comes up in several places is the check for FuncDecl/ConstructorDecl/VarDecl/SubscriptDecl followed by a call to replaceCovariantResultType() with the right uncurry level. But we can clean that up later.

Reply:
After a few attempts I was unable to adapt `TypeBase::adjustSuperclassMemberDeclType()` for  properties and subscripts.